recipes-security/optee/optee-os/0001-FIO-Trusted-Application.patch

```diff
@@ -0,0 +1,546 @@
++			       TEE_DATA_FLAG_ACCESS_WRITE |
++			       TEE_DATA_FLAG_OVERWRITE;
++	TEE_Result res;
++	TEE_ObjectHandle h;
++
++	char name_full[TEE_OBJECT_ID_MAX_LEN];
++	uint32_t name_full_sz;
++
++	if (pt != exp_pt)
++		return TEE_ERROR_BAD_PARAMETERS;
++
++	char *name_buf = params[0].memref.buffer;
++	uint32_t name_buf_sz = params[0].memref.size;
++
++	char *value = params[1].memref.buffer;
++	uint32_t value_sz = params[1].memref.size;
++
++	res = get_named_object_name(name_buf, name_buf_sz,
++				    name_full, &name_full_sz);
++	if (res)
++		return res;
++
++	res = TEE_CreatePersistentObject(storageid, name_full,
++					 name_full_sz,
++					 flags, NULL, value,
++					 value_sz, &h);
++	if (res)
++		EMSG("Can't create named object value, res = 0x%x", res);
++
++	TEE_CloseObject(h);
++
++	return res;
++}
++
++static TEE_Result read_persist_value(uint32_t pt,
++				      TEE_Param params[TEE_NUM_PARAMS])
++{
++	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
++						TEE_PARAM_TYPE_MEMREF_INOUT,
++						TEE_PARAM_TYPE_NONE,
++						TEE_PARAM_TYPE_NONE);
++	uint32_t flags = TEE_DATA_FLAG_ACCESS_READ |
++			 TEE_DATA_FLAG_ACCESS_WRITE;
++	TEE_Result res;
++	TEE_ObjectHandle h;
++
++	char name_full[TEE_OBJECT_ID_MAX_LEN];
++	uint32_t name_full_sz;
++	uint32_t count;
++
++	if (pt != exp_pt)
++		return TEE_ERROR_BAD_PARAMETERS;
++
++	char *name_buf = params[0].memref.buffer;
++	uint32_t name_buf_sz = params[0].memref.size;
++
++	char *value = params[1].memref.buffer;
++	uint32_t value_sz = params[1].memref.size;
++
++	res = get_named_object_name(name_buf, name_buf_sz,
++				    name_full, &name_full_sz);
++	if (res)
++		return res;
++
++	res = TEE_OpenPersistentObject(storageid, name_full,
++				       name_full_sz, flags, &h);
++	if (res) {
++		EMSG("Can't open named object value, res = 0x%x", res);
++		return res;
++	}
++
++	res =  TEE_ReadObjectData(h, value, value_sz, &count);
++	if (res) {
++		EMSG("Can't read named object value, res = 0x%x", res);
++		goto out;
++	}
++
++	params[1].memref.size = count;
++out:
++	TEE_CloseObject(h);
++
++	return res;
++}
++
++TEE_Result TA_CreateEntryPoint(void)
++{
++	return TEE_SUCCESS;
++}
++
++void TA_DestroyEntryPoint(void)
++{
++}
++
++TEE_Result TA_OpenSessionEntryPoint(uint32_t pt __unused,
++				    TEE_Param params[4] __unused,
++				    void **session __unused)
++{
++	return TEE_SUCCESS;
++}
++
++void TA_CloseSessionEntryPoint(void *sess __unused)
++{
++}
++
++TEE_Result TA_InvokeCommandEntryPoint(void *sess __unused, uint32_t cmd,
++				      uint32_t pt,
++				      TEE_Param params[TEE_NUM_PARAMS])
++{
++	switch (cmd) {
++	case TA_FIOVB_CMD_READ_ROLLBACK_INDEX:
++		return read_rb_idx(pt, params);
++	case TA_FIOVB_CMD_WRITE_ROLLBACK_INDEX:
++		return write_rb_idx(pt, params);
++	case TA_FIOVB_CMD_READ_LOCK_STATE:
++		return read_lock_state(pt, params);
++	case TA_FIOVB_CMD_WRITE_LOCK_STATE:
++		return write_lock_state(pt, params);
```

Comment:
You can drop the read/write lock state as that will not be used by us. This will also reduce a bit the side of the patch.